[2024-12-26T06:16:28.617+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-26T06:16:28.921+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T14:00:00+00:00 [queued]>
[2024-12-26T06:16:29.111+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T14:00:00+00:00 [queued]>
[2024-12-26T06:16:29.114+0000] {taskinstance.py:2866} INFO - Starting attempt 3 of 2
[2024-12-26T06:16:29.608+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): fetch_and_produce_data> on 2024-12-25 14:00:00+00:00
[2024-12-26T06:16:31.323+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=217) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-26T06:16:31.009+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'api_to_kafka_pipeline', 'fetch_and_produce_data', 'scheduled__2024-12-25T14:00:00+00:00', '--job-id', '75', '--raw', '--subdir', 'DAGS_FOLDER/api_to_kafka_dag.py', '--cfg-path', '/tmp/tmp68aepf4p']
[2024-12-26T06:16:31.418+0000] {standard_task_runner.py:72} INFO - Started process 274 to run task
[2024-12-26T06:16:31.507+0000] {standard_task_runner.py:105} INFO - Job 75: Subtask fetch_and_produce_data
[2024-12-26T06:16:35.513+0000] {task_command.py:467} INFO - Running <TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T14:00:00+00:00 [running]> on host a7d7ef0c0878
[2024-12-26T06:16:37.233+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='api_to_kafka_pipeline' AIRFLOW_CTX_TASK_ID='fetch_and_produce_data' AIRFLOW_CTX_EXECUTION_DATE='2024-12-25T14:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-25T14:00:00+00:00'
[2024-12-26T06:16:37.319+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-26T06:16:38.609+0000] {logging_mixin.py:190} INFO - âœ… Data fetched successfully for IBM at 5min interval.
[2024-12-26T06:16:38.614+0000] {logging_mixin.py:190} INFO - ðŸš€ Sending data to Kafka...
[2024-12-26T12:29:27.271+0000] {job.py:229} INFO - Heartbeat recovered after 20485.26 seconds
[2024-12-26T13:54:42.559+0000] {local_task_job_runner.py:228} ERROR - Heartbeat time limit exceeded!
[2024-12-26T13:54:42.753+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-12-26T13:54:43.057+0000] {process_utils.py:132} INFO - Sending 15 to group 274. PIDs of all processes in the group: [274]
[2024-12-26T13:54:43.078+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 274
[2024-12-26T13:55:43.193+0000] {process_utils.py:150} WARNING - process psutil.Process(pid=274, name='airflow task ru', status='sleeping', started='06:16:29') did not respond to SIGTERM. Trying SIGKILL
[2024-12-26T13:55:43.203+0000] {process_utils.py:87} INFO - Sending the signal 9 to group 274
[2024-12-26T13:55:43.231+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=274, name='airflow task ru', status='terminated', exitcode=<Negsignal.SIGKILL: -9>, started='06:16:29') (274) terminated with exit code -9
[2024-12-26T13:55:43.235+0000] {standard_task_runner.py:190} ERROR - ('Job 75 was killed before it finished (likely due to running out of memory)', 'For more information, see https://airflow.apache.org/docs/apache-airflow/stable/troubleshooting.html#LocalTaskJob-killed')
